ports/geckolib/gecko_bindings/sugar/ns_style_coord.rs

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use bindings::{Gecko_ResetStyleCoord, Gecko_SetStyleCoordCalcValue, Gecko_AddRefCalc};
+use std::mem::transmute;
+use structs::{nsStyleCoord_CalcValue, nsStyleCoord_Calc, nsStyleUnit, nsStyleUnion, nsStyleCoord};
+
+// Functions here are unsafe because it is possible to use the wrong nsStyleUnit
+// FIXME we should be pairing up nsStyleUnion and nsStyleUnit somehow
+// nsStyleCoord is one way to do it, but there are other structs using pairs
+// of union and unit too
+
+impl nsStyleUnion {
+    /// Clean up any resources used by an nsStyleUnit
+    /// Currently, this only happens if the nsStyleUnit
+    /// is a Calc
+    pub unsafe fn reset(&mut self, unit: &mut nsStyleUnit) {
+        if *unit == nsStyleUnit::eStyleUnit_Calc {
+            Gecko_ResetStyleCoord(unit, self);
+        }
+    }
+
+    /// Set internal value to a calc() value
+    /// reset() the union before calling this
+    pub unsafe fn set_calc_value(&mut self, unit: &mut nsStyleUnit, v: nsStyleCoord_CalcValue) {
+        // Calc should have been cleaned up
+        assert!(*unit != nsStyleUnit::eStyleUnit_Calc);
+        Gecko_SetStyleCoordCalcValue(unit, self, v);
+    }
+
+    pub unsafe fn get_calc(&self) -> nsStyleCoord_CalcValue {
+        (*self.as_calc())._base
+    }
+
+    pub unsafe fn addref_opt(&mut self, unit: &nsStyleUnit) {
```

Comment:
What does the _opt mean here? Optional? Seems kinda ambiguous.

But also, maybe this is the wrong level of abstraction? Shouldn't we just have a set() for nsStyleUnion that does the refcount munging as appropriate, and possibly a set_no_calc that asserts against calc in the cases we know it can't happen?